src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -558,15 +569,40 @@ private[fastq] object FastqDemultiplexer {
   /** Stores the minimal information for a single template read. */
   case class DemuxRecord(name: String, bases: String, quals: String, molecularBarcode: Option[String],
                          sampleBarcode: Option[String], readNumber: Int, pairedEnd: Boolean, comment: Option[String],
-                         originalBases: Option[String] = None, originalQuals: Option[String] = None)
+                         originalBases: Option[String] = None, originalQuals: Option[String] = None) {
+
+    /** A function to mask bases that have a quality score less than or equal to a specified threshold
+      * @param threshold The threshold for masking bases, inclusive. Bases with a quality score <= threshold will be masked
+      * @param qualityEncoding The encoding used for quality scores in the Fastq file.
+      * @return a new DemuxRecord with updated bases
+      */
+    def maskLowQualityBases(threshold: Char, qualityEncoding: QualityEncoding): DemuxRecord = {
+      if (this.quals.exists(_ <= threshold)) {
+        val bases = this.bases.toCharArray
+        for (i <- Range(0, this.quals.length)) if (quals.charAt(i) <= threshold) bases(i) = 'N'
+        this.copy(bases = bases.mkString)
+      }
+      else this
```

Comment:
1.  The usage for DemuxFastqs says "Mask bases with a quality score below the specified threshold as Ns".  But this implements `<=`.  Please update the function usages please as I prefer `<` versus `<=`.
2. Can you add a short-circuit so that if `threshold` is zero or below, you just return `this`
3. The suggestion below incorporates (1) & (2), and I prefer to have the shorter conditional branch first, as then it's much easer to scan.
```suggestion
      if (threshold <= 0 || this.quals.forall(_ > threshold) this else {
        val bases = this.bases.toCharArray
        for (i <- Range(0, this.quals.length)) if (quals.charAt(i) <= threshold) bases(i) = 'N'
        this.copy(bases = bases.mkString)
      }
```